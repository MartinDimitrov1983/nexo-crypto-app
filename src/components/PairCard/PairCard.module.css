.cardWrapper {
    display: flex;
    align-items: center;
    margin-bottom: 10px;
    border-radius: 5px;
    background-color: #fff;
    padding: 10px 10px;
    filter: drop-shadow(0 0 0.2rem rgb(193, 193, 193));
    width: 450px;
}

@media only screen and (max-width: 560px) {
    .cardWrapper {
        width: 250px;
    }
}

.cardWrapper:hover {
    transform: scale(1.03);
}

.mainIcon {
    width: 30px;
    height: 30px;
    padding: 10px;
}

.cardContent {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    margin-left: 10px;
}

.name {
    display: flex;
    justify-content: space-between;
    color: #5b6064;
    font-weight: 800;
    text-transform: capitalize;
    cursor: default;
}

.info {
    display: flex;
    margin-bottom: 0;
    padding-left: 0;
    font-size: 16px;
    cursor: default;
}

.pointer {
    cursor: pointer;
}
