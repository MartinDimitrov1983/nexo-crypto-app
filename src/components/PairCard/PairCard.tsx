import React, { MouseEventHandler } from 'react';
import styles from './PairCard.module.css';
import { Kraken, Bitfinex, Huobi, Binance } from '../../assets';
import { INFO_NOT_AVAILABLE, pricePair } from '../../constants';
import { TRADING_PROVIDERS } from '../../types';

interface PairCardProps {
    name: string;
    pair: string | undefined;
    price: number;
    onClick?: MouseEventHandler<HTMLDivElement>;
}

export const EXCHANGES = {
    [TRADING_PROVIDERS.binance]: <Binance className={styles.mainIcon} />,
    [TRADING_PROVIDERS.bitfinex]: <Bitfinex className={styles.mainIcon} />,
    [TRADING_PROVIDERS.huobi]: <Huobi className={styles.mainIcon} />,
    [TRADING_PROVIDERS.kraken]: <Kraken className={styles.mainIcon} />,
};

const PairCard = ({ name, pair, price, onClick = () => {} }: PairCardProps) => {
    return (
        <div className={styles.cardWrapper}>
            {EXCHANGES[name as keyof typeof EXCHANGES]}

            <div className={styles.cardContent}>
                <div className={styles.name}>{name}</div>

                <div
                    className={`${styles.info} ${price && styles.pointer}`}
                    onClick={price ? onClick : () => {}}
                >
                    {price ? pricePair(pair, price) : INFO_NOT_AVAILABLE}
                </div>
            </div>
        </div>
    );
};

export default PairCard;
