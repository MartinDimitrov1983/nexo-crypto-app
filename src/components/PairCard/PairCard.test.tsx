import React from 'react';
import { render } from '@testing-library/react';

import PairCard from './PairCard';

test('Snapshot PairCard', () => {
    const { asFragment } = render(
        <PairCard name="test" pair="test" price={10} onClick={() => {}} />,
    );
    expect(asFragment()).toMatchSnapshot();
});
