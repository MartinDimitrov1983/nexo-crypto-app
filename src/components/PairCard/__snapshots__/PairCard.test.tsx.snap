// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot PairCard 1`] = `
<DocumentFragment>
  <div
    class="cardWrapper"
  >
    <div
      class="cardContent"
    >
      <div
        class="name"
      >
        test
      </div>
      <div
        class="info pointer"
      >
        1 TEST = 10
      </div>
    </div>
  </div>
</DocumentFragment>
`;
